### PR TITLE
[SPARK-11343] [ML] Allow float and double prediction/label columns in RegressionEvaluator

### PR DESCRIPTION
**Pull request by dahlem**
@mengxr, @felixcheung 

This pull request just relaxes the type of the prediction/label columns to be float and double. Internally, these columns are casted to double. The other evaluators might need to be changed also.

### CODE REVIEW

**Comment by mengxr** [comment]
ok to test

**Comment by mengxr** [comment]
LGTM pending Jenkins.

File: mllib/src/main/scala/org/apache/spark/ml/evaluation/RegressionEvaluator.scala [hunk 1]
```diff
@@ -72,10 +73,13 @@ final class RegressionEvaluator @Since("1.4.0") (@Since("1.4.0") override val ui
   @Since("1.4.0")
   override def evaluate(dataset: DataFrame): Double = {
     val schema = dataset.schema
-    SchemaUtils.checkColumnType(schema, $(predictionCol), DoubleType)
-    SchemaUtils.checkColumnType(schema, $(labelCol), DoubleType)
+    val predictionType = schema($(predictionCol)).dataType
+    require(predictionType == FloatType || predictionType == DoubleType)
```
Comment (felixcheung) [inline mllib/src/main/scala/org/apache/spark/ml/evaluation/RegressionEvaluator.scala]: should we add a message to require()?

**Comment by mengxr** [comment]
Merged into master. @dahlem Could you send a new PR to address @felixcheung 's comment?

**Comment by dahlem** [comment]
No problem. Do you want me to do it against the same ticket?

**Pull request closed without merging**